Tests/TuistGeneratorTests/GraphViz/GraphToGraphVizMapperTests.swift

```diff
@@ -52,25 +38,27 @@ final class GraphToGraphVizMapperTests: XCTestCase {
 
     func test_map_skipping_external_dependencies() throws {
         // Given
-        let project = Project.test()
-        let framework = FrameworkNode.test(path: AbsolutePath("/XcodeProj.framework"))
-        let library = LibraryNode.test(path: AbsolutePath("/RxSwift.a"))
-        let sdk = try SDKNode(name: "CoreData.framework", platform: .iOS, status: .required, source: .developer)
+        let graph = try makeGivenGraph()
 
-        let core = TargetNode.test(target: Target.test(name: "Core"), dependencies: [
-            framework, library, sdk,
-        ])
-        let iOSApp = TargetNode.test(target: Target.test(name: "Tuist iOS"), dependencies: [core])
-        let watchApp = TargetNode.test(target: Target.test(name: "Tuist watchOS"), dependencies: [core])
+        // When
+        let got = subject.map(graph: graph, skipTestTargets: false, skipExternalDependencies: true, targetsToFilter: [])
+        let expected = makeExpectedGraphViz(includeExternalDependencies: false)
+        let gotNodeIds = got.nodes.map(\.id).sorted()
+        let expectedNodeIds = expected.nodes.map(\.id).sorted()
+        let gotEdgeIds = got.edges.map { $0.from + " -> " + $0.to }.sorted()
+        let expectedEdgeIds = expected.edges.map { $0.from + " -> " + $0.to }.sorted()
 
-        let graph = Graph.test(entryNodes: [iOSApp, watchApp],
-                               projects: [project],
-                               precompiled: [framework, library],
-                               targets: [project.path: [core, iOSApp, watchApp]])
+        XCTAssertEqual(gotNodeIds, expectedNodeIds)
+        XCTAssertEqual(gotEdgeIds, expectedEdgeIds)
+    }
+
+    func test_map_filter_targets() throws {
+        // Given
+        let graph = try makeGivenGraph()
 
         // When
-        let got = subject.map(graph: graph, skipTestTargets: false, skipExternalDependencies: true)
-        let expected = makeExpectedGraphViz(includeExternalDependencies: false)
+        let got = subject.map(graph: graph, skipTestTargets: false, skipExternalDependencies: true, targetsToFilter: ["Tuist iOS"])
```

Comment:
would this test fail if the targets actually wouldn't be filtered? afaict it would not but maybe I am mistaken. Might be worth creating a target alongside `Tuist iOS`

Reply:
It should fail because of the “tuist watchos” which is at the same level as “tuist iOS”, right?

Reply:
ah, OK, I got sidetrack by `onlyiOS` below but now it makes sense, sorry 👍